include/sgx/faasm_sgx_enclave_types.h

```diff
@@ -24,13 +27,58 @@
     faasm_sgx_tcs[THREAD_ID] = _FAASM_SGX_TCS_SLOT_FREE;                       \
     read_unlock(&_rwlock_faasm_sgx_tcs_realloc);
 
+#if(FAASM_SGX_ATTESTATION)
```

Comment:
It would be nice to reduce the number of directives we use. Is there any harm in removing this (i.e. defining the structs regardless of whether attestation is turned on?).

The reason I'd like to reduce the number of directives is that they result in dead code that isn't tested or compiled regularly. This means things can rot very quickly.

Reply:
I will remove that directive and simply assume everybody wants to use attestation when using SGX. We can implement a fallback to SGX execution without attestation, e.g. when the keymanager is unreachable

Reply:
Nice :+1:

Reply:
done :heavy_check_mark:  see dfcfc18

Reply:
auto-fallback to non-attestation mode is not implemented, though. I'll leave that for a future PR